plugins/usbpro/OpenDeckDevice.cpp

```diff
@@ -0,0 +1,256 @@
+    m_mab_time = params.mab_time;
+    m_rate = params.rate;
+  }
+}
+
+
+/*
+ * Handle a parameter request. This may set some parameters in the widget.
+ * If no parameters are set we simply fetch the parameters and return them to
+ * the client. If we are setting parameters, we send a SetParam() request and
+ * then another GetParam() request in order to return the latest values to the
+ * client.
+ */
+void OpenDeckDevice::HandleParametersRequest(RpcController *controller,
+                                             const Request *request,
+                                             string *response,
+                                             ConfigureCallback *done) {
+  if (request->has_parameters() &&
+      (request->parameters().has_break_time() ||
+       request->parameters().has_mab_time() ||
+       request->parameters().has_rate())) {
+    if (!m_got_parameters) {
+      controller->SetFailed("SetParameters failed, startup not complete");
+      done->Run();
+      return;
+    }
+
+    bool ret = m_opendeck_widget->SetParameters(
+      request->parameters().has_break_time() ?
+        request->parameters().break_time() : m_break_time,
+      request->parameters().has_mab_time() ?
+        request->parameters().mab_time() : m_mab_time,
+      request->parameters().has_rate() ?
+        request->parameters().rate() : m_rate);
+
+    if (!ret) {
+      controller->SetFailed("SetParameters failed");
+      done->Run();
+      return;
+    }
+  }
+
+  m_opendeck_widget->GetParameters(NewSingleCallback(
+    this,
+    &OpenDeckDevice::HandleParametersResponse,
+    controller,
+    response,
+    done));
+}
+
+
+/**
+ * Handle the GetParameters response
+ */
+void OpenDeckDevice::HandleParametersResponse(
+    RpcController *controller,
+    string *response,
+    ConfigureCallback *done,
+    bool status,
+    const usb_pro_parameters &params) {
+  if (!status) {
+    controller->SetFailed("GetParameters failed");
+  } else {
+    UpdateParams(true, params);
+    Reply reply;
+    reply.set_type(ola::plugin::usbpro::Reply::USBPRO_PARAMETER_REPLY);
+    ola::plugin::usbpro::ParameterReply *parameters_reply =
+      reply.mutable_parameters();
+
+    parameters_reply->set_firmware_high(params.firmware_high);
+    parameters_reply->set_firmware(params.firmware);
+    parameters_reply->set_break_time(params.break_time);
+    parameters_reply->set_mab_time(params.mab_time);
+    parameters_reply->set_rate(params.rate);
+    reply.SerializeToString(response);
+  }
+  done->Run();
+}
+
+
+/*
+ * Handle a Serial number Configure RPC. We can just return the cached number.
+ */
+void OpenDeckDevice::HandleSerialRequest(
+    RpcController *controller,
+    const Request *request,
+    string *response,
+    ConfigureCallback *done) {
+  Reply reply;
+  reply.set_type(ola::plugin::usbpro::Reply::USBPRO_SERIAL_REPLY);
+  ola::plugin::usbpro::SerialNumberReply *serial_reply =
+    reply.mutable_serial_number();
```

Comment:
Fix the indent:
```suggestion
      reply.mutable_serial_number();
```

Reply:
Its not applied?

Reply:
Weird, was sure that I did... Fixed now.